framework/config/src/org/apache/cloudstack/framework/config/ConfigKey.java

```diff
@@ -122,6 +122,13 @@ public boolean equals(Object obj) {
         if (obj instanceof ConfigKey) {
             ConfigKey<?> that = (ConfigKey<?>)obj;
             return this._name.equals(that._name);
+        }
+        return false;
+    }
+
+    public boolean isSameKeyAs(Object obj) {
+        if(this.equals(obj)) {
+            return true;
         } else if (obj instanceof String) {
```

Comment:
@DaanHoogland we shouldn't make assumptions about the loading behavior of a class.  We should write our equals implementations defensively.  Another issue with instanceof is that it returns true for compatible subclasses which is also incorrect from an equals perspective.